keystone/common/api_filtering.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2015 IBM Corp.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import copy
+from keystone import config
+from keystone import exception
+from keystone.models import token_model
+from keystone.i18n import _LE
+from oslo_config import cfg
+from oslo_log import log
+from oslo_log import loggers
+from functools import wraps
+
+LOG = log.getLogger(__name__)
+
+opt_group = cfg.OptGroup(name='api_filtering',
+                              title='Keystone API filtering')
+opts = [
+    cfg.ListOpt('projects_to_filter',
+                default=['admin', 'service', 'monitoring', 'support'],
+                help='List of projects to filter by default.'),
+    cfg.ListOpt('roles_to_filter',
+                default=['admin', 'KeystoneAdmin',
+                         'KeystoneServiceAdmin', 'service',
+                         'heat_domain_admin', 'support'],
+                help='List of roles to filter by default.'),
+    cfg.ListOpt('explicit_users_to_filter',
+                default=['heat_domain_admin'],
+                help='List of users to filter by default.'),
+    cfg.ListOpt('admin_roles',
+                default=['cloud_admin', 'project_admin'],
+                help='List of admin roles by default.')]
+
+CONF = config.CONF
+CONF.register_group(opt_group)
+CONF.register_opts(opts, opt_group)
```

Comment:
Can we move these to the keystone/common/config.py That way we have all the config options in one place rather than spread out over the code.